src/App.tsx

```diff
@@ -8,7 +8,54 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos, getUser } from './api';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [isOpenedModal, setIsOpenedModal] = useState(false);
+  const [idTodo, setIdTodo] = useState(0);
+  const [titleTodo, setTitleTodo] = useState('');
+  const [user, setUser] = useState<User | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const filtredByQuery = (items: Todo[]) => {
+    return items.filter(item => item.title.includes(query));
+  };
+
+  const filteredTodos = (items: Todo[]) => {
+    if (filter === 'active') {
+      const filtredItems = items.filter(item => !item.completed);
+
+      return filtredByQuery(filtredItems);
+    }
+
+    if (filter === 'completed') {
+      const filtredItems = items.filter(item => item.completed);
+
+      return filtredByQuery(filtredItems);
+    }
+
+    return filtredByQuery(items);
+  };
+
+  useEffect(() => {
+    getTodos().then(response => setTodos(response));
+  }, []);
+
+  const onOpenModal = (
+    value: boolean, todoId: number, userId: number, title: string,
+  ) => {
+    setIsOpenedModal(value);
+    setIdTodo(todoId);
+    setTitleTodo(title);
+
+    getUser(userId).then(person => setUser(person));
```

Comment:
you use it only inside `TodoModal` - move this state to this component (load in useEffect)